Sources/Articles/Documentation.docc/2024/swiftpm-snippets/SwiftPM Snippets.md

```diff
@@ -0,0 +1,264 @@
+
+@Code(name: "SnippetsExample_II.swift", file: SnippetsExample_II.swift)
+
+When embedded, it should look like this:
+
+---
+
+@Snippet(id: SnippetsExample_II)
+
+>   Note:
+>   Snippet captions cannot embed other Snippets.
+
+
+## Redacting parts of a Snippet
+
+You can redact portions of a Snippet using **slice directives**. A slice directive is a line comment token that starts with `snippet` followed by a dot and an identifier. The following identifiers have special meanings:
+
+| Identifier        | Behavior                                   |
+| ----------------- | ------------------------------------------ |
+| `snippet.hide`    | Hides the content following the directive. |
+| `snippet.end`     | Hides the content following the directive. |
+| `snippet.show`    | Shows the content following the directive. |
+
+
+Below is an example of a Snippet that uses redactions to hide the `import` statements.
+
+@Code(name: SnippetsExample_III.swift, file: SnippetsExample_III.swift)
+
+When embedded, it should look like this:
+
+---
+
+@Snippet(id: SnippetsExample_III)
+
+
+>   Warning:
+>   At the time of writing, there is a [known bug](https://github.com/apple/swift-docc/issues/946) preventing DocC from slicing Snippets that contain multi-line string literals correctly.
+
+
+### Slice indentation
+
+The indentation of the first `snippet.show` determines specifies the maximum amount of indentation to remove from the Snippet. In the example below, four spaces of indentation will be removed from the rendered Snippet. Note that the `snippet.end` token is required in order to prevent the Snippet from including the final brace, which would have prevented the indentation from being removed.
+
+@Code(name: SnippetsExample_IV.swift, file: SnippetsExample_IV.swift)
+
+When embedded, it should look like this:
+
+---
+
+@Snippet(id: SnippetsExample_IV)
+
+
+## Using named slices
+
+You can also use **named slices** to create Snippets with multiple embeddable sections. It’s a good idea to give slices uppercase names to distinguish them from special slice directives.
+
+Below is an example of a Snippet with a caption and three named slices.
+
+@Code(name: SnippetsExample_V.swift, file: SnippetsExample_V.swift)
+
+Here’s how you might embed the slices in a Markdown article.
+
+@Code(name: "My article.md", file: "My article (6).md.txt")
+
+And here’s how the embedded slices should look.
+
+---
+
+@Snippet(id: SnippetsExample_V, slice: DECLARATION)
+@Snippet(id: SnippetsExample_V, slice: BODY)
+@Snippet(id: SnippetsExample_V, slice: EXIT)
+
+
+## Rendering Snippets with linkable references
+
+Some documentation engines such as Unidoc support rendering Snippets with linked identifiers. This is possible because Snippets are compiled along with the rest of the package, allowing them to be mapped by [IndexStoreDB](https://github.com/apple/indexstore-db).
```

Comment:
It's a bit strange, that here we start mentioning Unidoc again, but so far only DocC examples were provided, why don't we create a standalone follow-up article about using Unidoc & snippets and keep this one simple (Snippets + DocC). 🤔

Reply:
i rewrote the entire section as a **Where to go from here** and removed references to IndexStoreDB